.github/workflows/code-checks.yml

```diff
@@ -26,17 +32,20 @@ jobs:
       uses: actions/cache@v3
       with:
         path: ~/.cache/pip
-        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/pytorch/base.txt') }}
+        key: ${{ runner.os }}-pip-${{ matrix.name }}-${{ hashFiles('requirements/pytorch/base.txt') }}
         restore-keys: |
           ${{ runner.os }}-pip-
 
-    - name: Install dependencies
+    - name: Adjust dependencies
+      if: matrix.name == "pytorch"
       run: |
         pip install torch==1.12 --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
         python ./requirements/pytorch/adjust-versions.py requirements/pytorch/extra.txt
-        # todo: adjust requirements for both code-bases
-        pip install -r requirements/pytorch/devel.txt --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
+
+    - name: Install dependencies
+      run: |
+        pip install -r requirements/${{ matrix.name }}/devel.txt --find-links https://download.pytorch.org/whl/cpu/torch_stable.html
         pip list
 
     - name: Check typing
-      run: mypy
+      run: mypy ./src/${{ matrix.pkg }}
```

Comment:
Needs to uncomment https://github.com/Lightning-AI/lightning/blob/7a617ec90e1566c763be8ac7a200af1e4025412c/pyproject.toml#L29-L30

Reply:
well, the point is to run it separately or do you suggest running always all?

Reply:
I think this job could run the specific package (as you did)

and for contributors, `$ mypy` runs all